src/cloudant/index.py

```diff
@@ -262,3 +265,79 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
```

Comment:
Are you planning on deleting this class and moving the codifying of the index function to the DesignDocument like we discussed?